mmtk/src/active_plan.rs

```diff
@@ -71,29 +84,23 @@ impl ActivePlan<JuliaVM> for VMActivePlan {
     }
 }
 
-#[no_mangle]
-pub extern "C" fn get_next_mutator_tls() -> OpaquePointer {
-    let mutators = MUTATORS.read().unwrap();
-
-    let mutator_idx = unsafe { get_next_julia_mutator() };
-    let mutator = mutators.get(mutator_idx);
-
-    let res = match mutator {
-        Some(m) => {
-            let mutator = unsafe { get_mutator_from_ref(*m) };
+// Expose the mutator iterator so they can be used in C.
 
-            unsafe { (*mutator).mutator_tls.0 .0 }
-        }
-        None => {
-            unsafe { reset_mutator_count() }
-            OpaquePointer::from_address(unsafe { Address::zero() })
-        }
-    };
+#[no_mangle]
+pub extern "C" fn new_mutator_iterator() -> Box<JuliaMutatorIterator<'static>> {
+    let guard = MUTATORS.read().unwrap();
+    Box::new(JuliaMutatorIterator::new(guard))
+}
 
-    res
+#[no_mangle]
+pub extern "C" fn get_next_mutator_tls(iter: &mut JuliaMutatorIterator<'static>) -> OpaquePointer {
+    match iter.next() {
+        Some(m) => m.mutator_tls.0 .0,
+        None => OpaquePointer::from_address(Address::ZERO),
+    }
 }
 
 #[no_mangle]
-pub extern "C" fn reset_count_tls() {
-    unsafe { reset_mutator_count() }
+pub extern "C" fn close_mutator_iterator(_iter: Box<JuliaMutatorIterator<'static>>) {
+    // Do nothing and the iterator will be dropped.
```

Comment:
```suggestion
pub extern "C" fn close_mutator_iterator(iter: *mut JuliaMutatorIterator<'static>) {
    unsafe { Box::from_raw(iter) };
    // Do nothing and the iterator will be dropped.
```

Similarly, we can turn it back into a `Box`, and it will be automatically dropped.